File: LL/crud.cpp
#include<bits/stdc++.h>
#include<iostream>

using namespace std;

class Node{
  
  public:
  int data;
  Node* next;

  Node (int data1, Node* next1 = nullptr){
    data = data1;
    next = next1;
  }
};

Node *head;

void traverse(){
  Node *ptr = head;
  while(ptr){
    cout << ptr->data;
    ptr = ptr->next;
  }
}

void insertAtbeginning(){
  Node *p = new Node(5,head);
  head = p;
  cout << "\n";
  //return p;
  traverse();
  cout << "\n";
}

void insertAtEnd(){
  Node *p = new Node(15,nullptr);
  Node *ptr = head;
  while(ptr->next != nullptr){
    ptr = ptr->next;
  }

  ptr->next = p;
  cout << "\n";
  //return p;
  traverse();
  cout << "\n";
}

int main(){
  Node *p = new Node(10);
  cout << p->next;
  head = p;
  insertAtbeginning();
  insertAtEnd();
  cout << head->data;

  return 0;
}

File: LL/crud.py
class Node:
    def __init__(self,data=None,next=None):
        self.data = data
        self.next = next

class LL:
    def __init__(self):
       self.head = None

    def insert_at_beginning(self,data):
        newNode = Node(data,self.head)
        self.head = newNode
    
    def insert_at_back(self,data):
        newNode = Node(data,None)
        
        temp = self.head
        while(temp.next != None):
            temp = temp.next
        
        temp.next = newNode

    def remove_from_front(self):
        temp = self.head
        self.head = self.head.next

    def remove_from_end(self):
        temp = self.head
        while(temp.next.next != None):
            #print(temp.data,end=" -> ")
            temp = temp.next
        
        temp.next = None

    def insert_anywhere(self,pos,data):
        if pos == 0:
            self.insert_at_beginning(data)
            return
        temp = self.head
        for i in range(pos-1):
            temp = temp.next
        temp1 = temp.next
        if not temp1:
            self.insert_at_back(data)
            return
        n = Node(data)
        temp.next = n
        n.next = temp1


    def delete_anywhere(self,pos):
        if pos == 0:
            self.remove_from_front()
            return
        temp = self.head
        for i in range(pos-1):
            temp = temp.next
        temp1 = temp.next
        if not temp1:
            self.remove_from_end()    
            return
        temp.next = temp.next.next
        temp1.next = None



    def printLL(self):
        temp = self.head
        while(temp):
            print(temp.data,end=" -> ")
            temp = temp.next
        print("None")
        
ll = LL()
ll.insert_at_beginning(10)
ll.insert_at_back(15)
ll.insert_at_beginning(5)
ll.remove_from_front()
ll.insert_anywhere(2,20)
ll.delete_anywhere(2)
ll.printLL()

    